### DISCUSSION TITLE
Updating a 3.x syntax gulp.js file to 4.x

### DISCUSSION THREAD

**Discussion by itanimo** [gulpjs/gulp#2520] [Help] [answered]
Hey there, can anyone help me how to update the syntax on the following file to be 4.x compatible? I'm new to gulp and I'm doing a test project that includes it. Here's the content:

```
var gulp = require('gulp');
var ts = require('gulp-typescript');
var tsProject = ts.createProject('tsconfig.json');

gulp.task('default', ['move-dev-config', 'move-test-config'], function() {
  return tsProject.src()
    .pipe(tsProject())
    .js.pipe(gulp.dest('built'));
});

gulp.task('move-dev-config', function() {
  return gulp.src(['./src/test-config.json', './src/dev-config.json'])
    .pipe(gulp.dest('built'));
});

gulp.task('move-test-config', function() {
  return gulp.src(['./src/test/mocha.opts'])
    .pipe(gulp.dest('built/test'));
});

gulp.task('scripts', function() {
  return tsProject.src()
    .pipe(tsProject())
    .js.pipe(gulp.dest('built'));
});

gulp.task('watch', ['scripts'], function() {
  gulp.watch([
    './src/*.ts',
    './src/**/*.ts',
    './src/**/**/*.ts',
  ], ['scripts']);
});
```
Any time I compile with the above I get AssertionErrors and Tasks Not defined.

**Comment by moatefdev** [comment]
### As far as I know one of the differences between gulp 3 and gulp 4 is the following:

You will use [**series()**](https://gulpjs.com/docs/en/api/series/)  and pass the task name to it, instead of putting it between **[]**  brackets.

**In gulp 3:**
`gulp.task('watch', ['scripts'], function() {
  gulp.watch([
    './src/*.ts',
    './src/**/*.ts',
    './src/**/**/*.ts',
  ], ['scripts']);
});`

**In gulp 4:**
`gulp.task('watch', ['scripts'], function() {
  gulp.watch([
    './src/*.ts',
    './src/**/*.ts',
    './src/**/**/*.ts',
  ], gulp.series('scripts'));
});`

![carbon](https://user-images.githubusercontent.com/53702629/101626536-c815b980-3a25-11eb-99f2-6d510f7ffb62.png)

**Reply by yocontra** [reply]
You still included the task array in your gulp 4 example which won't work

**Comment by yocontra** [comment]
You can find guides on how to upgrade around the internet: 

- https://www.sitepoint.com/how-to-migrate-to-gulp-4/
- https://wildermuth.com/2019/10/28/Upgrading-from-Gulp-3-to-4

**Comment by phated** [comment, marked as answer]
It's highly recommended to read through our entire getting started guide. We put a ton of effort into it so everyone can know the best patterns to use for their gulp 4 setups. https://gulpjs.com/docs/en/getting-started/quick-start
